location/src/iosMain/kotlin/location/LocationManager.kt

```diff
@@ -0,0 +1,142 @@
+            autoEnableLocations: Boolean,
+            locationStateRepo: LocationStateRepo
+        ): BaseLocationManager = LocationManager(
+            locationManager,
+            locationPermission,
+            permissions,
+            autoRequestPermission,
+            autoEnableLocations,
+            locationStateRepo
+        )
+    }
+
+    private val locationManagerDelegate = object : NSObject(), CLLocationManagerDelegateProtocol {
+
+        override fun locationManager(manager: CLLocationManager, didUpdateLocations: List<*>) {
+            if (isMonitoringLocationUpdate) {
+                val locations = didUpdateLocations.mapNotNull { (it as? CLLocation)?.knownLocation }
+                if (locations.isNotEmpty()) {
+                    handleLocationChanged(locations)
+                }
+            }
+        }
+
+        override fun locationManager(manager: CLLocationManager, didUpdateToLocation: CLLocation, fromLocation: CLLocation) {
+            if (isMonitoringLocationUpdate) {
+                handleLocationChanged(listOf(didUpdateToLocation.knownLocation))
+            }
+        }
+
+        override fun locationManager(manager: CLLocationManager, didFinishDeferredUpdatesWithError: NSError?) {
+        }
+
+        override fun locationManager(manager: CLLocationManager, didChangeAuthorizationStatus: CLAuthorizationStatus) {
+            if (isMonitoringLocationEnable) {
+                handleLocationEnabledChanged()
+            }
+        }
+    }
+
+    private val authorizationStatus: CLAuthorizationStatusKotlin
+        get() = Enum.byOrdinalOrDefault(
+            CLLocationManager.authorizationStatus(),
+            CLAuthorizationStatusKotlin.notDetermined
+        )
+
+    var isMonitoringLocationEnable = false
+    var isMonitoringLocationUpdate = false
+
+    override suspend fun startMonitoringLocationEnabled() {
+        isMonitoringLocationEnable = true
+        locationManager.delegate = locationManagerDelegate
+        locationManager.startUpdatingLocation()
+    }
+
+    override suspend fun stopMonitoringLocationEnabled() {
+        locationManager.stopUpdatingLocation()
+        isMonitoringLocationEnable = false
+    }
+
+    override suspend fun isLocationEnabled(): Boolean = when (authorizationStatus) {
+        // Enabled
+        CLAuthorizationStatusKotlin.authorizedAlways,
+        CLAuthorizationStatusKotlin.authorizedWhenInUse -> true
+        // Disabled
+        CLAuthorizationStatusKotlin.restricted,
+        CLAuthorizationStatusKotlin.denied,
+        CLAuthorizationStatusKotlin.notDetermined -> false
+    }
+
+    override suspend fun requestLocationEnable() {
+        // No access to UIApplication.openSettingsURLString
+        // We have to fallback to alert then?
+    }
+
+    override suspend fun startMonitoringLocation() {
+        isMonitoringLocationUpdate = true
+        locationManager.delegate = locationManagerDelegate
+        locationManager.startUpdatingLocation()
```

Comment:
Apart of this `startUpdatingLocation()` here is two more method (at least on iOS):
`startMonitoringSignificantLocationChanges()` and `startUpdatingHeading()` (and stop* as well)
And they're very useful depending or accuracy required in the app.
I think we have to create interface for them as well. Or make configuration which kind of updating required

Reply:
I wouldnt go this way as its not Android friendly. Better to provide filters via extension functions to Flow<Location> so its platform agnostic

Reply:
For `startMonitoringSignificantLocationChanges` filters would mean the underlying location gathering is still the more power intensive high accuracy method.

For `startUpdatingHeading` I think this is different, it combines compass with GPS I think? You might get more than just `startUpdatingLocation`?

I agree with @avdyushin it's useful to have, but let's get the base functionality first. Can you make an issue and add what you know @avdyushin ? (escp if I'm wrong about heading)

Reply:
For `startUpdatingLocation` you will get updates depends on `distanceFilter` value (5m, 10m), for `startMonitoringSignificantLocationChanges` it ignores it and do updates every 500m and for `startUpdatingLocation` you have extra delegate call with new `CLHeading` object for compass, it's not location but it's still in location manager, so can be useful. I'll create a new ticket for this

Reply:
Thanks make sure to put the info there so we can compare with Android, I think e.g. location with heading (which AFAIK is not just compass based but also accelorometer and location delta) is also available.